### DISCUSSION TITLE
Is a form block in core a possible future?

### DISCUSSION THREAD

**Discussion by Humanify-nl** [WordPress/gutenberg#39546] [Ideas]
So many times in the past 5 years working with Wordpress I have been wondering: “why is this not included in the standard Wordpress experience”

Most of the time I figured out the answer is: this is plugin territory, because people make their living of it. I suppose that could be called some sort of status quo? Or am I looking at this the wrong way?

With huge respect to the plug-in community, we may need some things in core. Because it belongs there. Now the block editor is getting really mature, it might open up the discussion of a form block. 

Forms have been integral to websites for a long time. The block API would be perfect for it, and many of the form elements are already there. What would be needed to make a form builder block happen? Somewhere along the line?

Is this a point of discussion at all?

I’d like to understand why not. I’m a long time GitHub lurker, and recently contributing more in small issues, but I may be overlooking some sensitivities here about forms, or the question what is plug-in territory and what is not.

**Comment by talldan** [comment]
I think there is a strong use case. I don't think the presentational aspect of forms are particularly challenging and the block editor would be a pretty great form builder. But there are a lot of technical challenges, particular around what forms do with their data.

Some may make an API request as part of submission, which isn't too difficult to support, but still a somewhat technical thing for a user to hook up.

Others trigger some JavaScript, which is a very complex thing to support. And further still users may want to store the data in their WordPress database or a third-party service. There are challenges with things like validation, security and privacy—each consumer of data or each different implementation of a form may have different requirements around those things. When you break it down, it's a pretty huge thing to implement!

Another aspect is looking at the journey that Gutenberg has been on, forms haven't really fit in anywhere yet. The first phase was all about replacing the post editor with a focus on writing, and more recently the journey has revolved around theming and full site editing. But that may change in the future, and perhaps there will be more opportunity in other phases to explore forms.

**Reply by Humanify-nl** [reply]
Thanks for your insights! It is a huge technical challenge indeed, although I’d think it could lay the groundwork and leave the advanced extensions to the (plug-in) developers.

But yes there are many other fantastic things being worked on that have priority, this would be way down the line.
